### PR TITLE
Changelog

### PR DESCRIPTION
**Pull request by jnevelson**
### Description

Thanks for contributing this Pull Request. Make sure that you send in this Pull Request to the `master` branch of this repository, add a brief description, and tag the relevant issue(s) and PR(s) below.

- Relevant Issues :

- What Changed and Why?
New functionality to update changelogs when sending out pull requests

- Type of change :
  - [x] New feature
  - [ ] Bug fix for existing feature
  - [ ] Code quality improvement
  - [ ] Addition or Improvement of tests
  - [ ] Addition or Improvement of documentation

- Other:
  - [ ] Add unit tests
  - [ ] Add documentation

### CODE REVIEW

File: gordian/files/changelog_file.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
+from . import MarkdownFile
+import re
+from datetime import datetime
+
+CHANGELOG_ENTRY_REGEX = '.*\[([0-9]*\.[0-9]*\.[0-9]*)\] - [0-9]{4}-[0-9]{1,2}-[0-9]+'
```
Comment (jnevelson) [inline gordian/files/changelog_file.py:5]: Right now this is not overridable, and I think it needs to be to work with different changelog formats. How should this is passed in? Config parameter?

Comment (AgarFu) [reply gordian/files/changelog_file.py:5]: add a complex regexp in the command line like this is kind of messy. We can leverage the configuration file maybe to add an attribute with this, but we can create an issue for that, maybe someone from the community can contribute with it.

Comment (jnevelson) [reply gordian/files/changelog_file.py:5]: Agreed - I wouldn't want it to be a CLI argument. I was imagining a Config object that could be configured with the regex.

File: gordian/files/changelog_file.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        lines = ['\n'.join(self.header)]
+        if entries:
+            lines.append('\n'.join(entries))
+            lines.append('')
+        lines.append('\n'.join(blocks))
+        lines.append('\n'.join(self.footer))
+
+        return '\n'.join(lines)
+
+    def _format_new_changelog_entry(self):
```
Comment (AgarFu) [inline gordian/files/changelog_file.py:77]: If we allow to change the entry regular expression, this method also needs to reflect that same structure.

File: tests/test_changelog_file.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        self.repo = Repo('test', git=self.mock_git)
+        self.repo.new_version = '1.2.0'
+        self.changelog = ChangelogFile(self.github_file, self.repo)
+
+    def test_iterable(self):
+        assert(iter(self.changelog))
+
+    def test_assert_added(self):
+        self.changelog.added('test')
+
```
Comment (AgarFu) [inline tests/test_changelog_file.py:22]: We need assertions here, what is happening when you call add?

File: tests/test_changelog_file.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    def test_assert_updated(self):
+        self.changelog.updated('test')
+
+    def test_assert_removed(self):
+        self.changelog.removed('test')
+
+    def test_save_changelog(self):
+        self.changelog.added('test')
+        self.changelog.save('save file', False)
+
```
Comment (AgarFu) [inline tests/test_changelog_file.py:32]: Same with the previous tests

File: gordian/files/changelog_file.py (hunk 2 above)
Comment (jnevelson) [reply gordian/files/changelog_file.py:77]: Yep, everything would need to be configurable. It'll end up being somewhat complex - almost more complex than Gordian itself.

**Pull request merged**